test/lint/lint-shell-locale.sh

```diff
@@ -16,7 +17,7 @@ for SHELL_SCRIPT in $(git ls-files -- "*.sh" | grep -vE "src/(secp256k1|univalue
         continue
     fi
     FIRST_NON_COMMENT_LINE=$(grep -vE '^(#.*|)$' "${SHELL_SCRIPT}" | head -1)
-    if [[ ${FIRST_NON_COMMENT_LINE} != "export LC_ALL=C" ]]; then
+    if [[ ${FIRST_NON_COMMENT_LINE} != "export LC_ALL=C" && ${FIRST_NON_COMMENT_LINE} != "export LC_ALL=C.UTF-8" ]]; then
```

Comment:
nit: `if [[ ! ${FIRST_NON_COMMENT_LINE} =~ "export LC_ALL=C"(|\.UTF-8)$ ]]; then`

Reply:
I prefer the current version. It allows for copy and paste.

Reply:
@ken2812221 @practicalswift 

what about

```
    case "${FIRST_NON_COMMENT_LINE}" in
      "export LC_ALL=C");;
      "export LC_ALL=C.UTF-8");;
      *)
        echo "Missing \"export LC_ALL=C\" (to avoid locale dependence) as first non-comment non-empty line in ${SHELL_SCRIPT}"
        EXIT_CODE=1
    esac
```

Reply:
LGTM

Reply:
pushed.

Reply:
@scravy Oh, I think a simple `if`-statement is much clearer than the suggested `case` construction. I suggest reverting back to the original version :-)

Reply:
🧐removed the commit again.

Reply:
@scravy Looks good now! Keep it as is.